File: f77/build05/discrete.f
C
      Subroutine Discrete
C
C       This subroutine processes the scheduled user-defined discrete events.
C
C       Inputs:
C           Coef_Of_Restitution     - Coefficent of restitution [Real]
C           Xd                                              - Velocity of ball [m/sec]
C
C       Outputs:
C           ReEval_Derivs   - Indicates states changed value at a discrete event
C                                                 [Logical]
C     Xd                              - Velocity of ball [m/sec]
C
C       Internal variables and constants:
C           None.
C
C       Declare global common
C
      Include '../driver/global.inc'
      Include '../driver/sysvars.inc'
      Include '../driver/event.inc'
      Equivalence (Real_Array(10), Coef_Of_Restitution)
      Equivalence (Real_Array(15), Xd)
C
C     Check for valid user event
C
      If (Events(1) .eq. 10) Then
         Xd = -Coef_Of_Restitution*Xd
      End If
      ReEval_Derivs = .TRUE.
C
      Return
      End

File: compcert/build05/discrete.c
#include "discrete.h"
#include "../driver/global.h"
#include "../driver/sysvars.h"
#include "../driver/event.h"

#define coef_of_restitution real_array[10].r  // Coefficent of restitution [Real]
#define xd                  real_array[15].r  // Velocity of ball [m/sec]
#define BOUNCE 10
void discrete() {
  // This subroutine processes the scheduled user-defined discrete events.
  // Outputs:
  //     ReEval_Derivs   - Indicates states changed value at a discrete event [Logical]
  //     Xd              - Velocity of ball [m/sec]
  // Check for valid user event
  if (BOUNCE == events[0])
    xd = -coef_of_restitution*xd;
  reeval_derivs = TRUE;
}
